packages/editor/src/components/post-author/index.js

```diff
@@ -17,34 +24,217 @@ export class PostAuthor extends Component {
+	 */
+	setAuthorId( selection ) {
+		if ( ! selection ) {
+			return;
+		}
 		const { onUpdateAuthor } = this.props;
-		const { value } = event.target;
-		onUpdateAuthor( Number( value ) );
+		if ( typeof selection === 'string' ) {
+			// Author name from the autocompleter.
+			const author = this.authors.find( ( singleAuthor ) => {
+				return this.authorUniqueName( singleAuthor ) === selection;
+			} );
+			if ( author ) {
+				this.setState( { currentSelection: selection } );
+				onUpdateAuthor( Number( author.id ) );
+			}
+		} else {
+			// Author ID from the select.
+			onUpdateAuthor( Number( selection.target.value ) );
+		}
 	}
 
 	render() {
-		const { postAuthor, instanceId, authors } = this.props;
-		const selectId = 'post-author-selector-' + instanceId;
+		const { postAuthor } = this.state;
+		const { postAuthorId, instanceId, authors } = this.props;
+		const selectId = `post-author-selector-${ instanceId }`;
+		let selector;
+		if ( ! postAuthor ) {
+			return null;
+		}
 
-		// Disable reason: A select with an onchange throws a warning
-
-		/* eslint-disable jsx-a11y/no-onchange */
-		return (
-			<PostAuthorCheck>
-				<label htmlFor={ selectId }>{ __( 'Author' ) }</label>
+		// If there are a small number of users, use a regular select element; once the list grows to a certain size, switch
+		// to an accessible auto-complete component.
+		if ( authors.length < 99 ) {
+			// Disable reason: A select with an onchange throws a warning.
+			/* eslint-disable jsx-a11y/no-onchange */
+			selector =
 				<select
 					id={ selectId }
-					value={ postAuthor }
+					value={ postAuthorId }
 					onChange={ this.setAuthorId }
 					className="editor-post-author__select"
 				>
 					{ authors.map( ( author ) => (
 						<option key={ author.id } value={ author.id }>{ decodeEntities( author.name ) }</option>
 					) ) }
-				</select>
+				</select>;
+		} else {
+			selector =
+				<Autocomplete
+					id={ selectId }
+					minLength={ 2 }
+					showAllValues={ true }
+					defaultValue={ postAuthor ? this.authorUniqueName( postAuthor ) : '' }
+					displayMenu="overlay"
+					onConfirm={ this.setAuthorId }
+					source={ this.suggestAuthor }
+					showNoOptionsFound={ false }
+					preserveNullOptions={ true }
+					tStatusQueryTooShort={ ( minQueryLength ) =>
+						// translators: %d: the number characters required to initiate an author search.
+						sprintf( __( 'Type in %d or more characters for results' ), minQueryLength )
+					}
+					tStatusNoResults={ () => __( 'No search results' ) }
+					// translators: 1: the index of thre selected result. 2: The total number of results.
+					tStatusSelectedOption={ ( selectedOption, length ) => sprintf( __( '%1$s (1 of %2$s) is selected' ), selectedOption, length ) }
+					tStatusResults={ ( length, contentSelectedOption ) => {
+						return (
+							_n( '%d result is available.', '%d results are available.', length ) +
+							' ' + contentSelectedOption
+						);
```

Comment:
As mentioned in the twin PR https://github.com/WordPress/gutenberg/pull/16666#pullrequestreview-272056920 I guess there should be a `sprintf()` here.